test/test_luaunit.lua

```diff
@@ -1137,6 +1164,33 @@ TestLuaUnitAssertions = { __class__ = 'TestLuaUnitAssertions' }
         lu.assertNotIsNumber( true)
     end
 
+    function TestLuaUnitAssertions:test_assertNotIsNaN()
+        lu.assertNotIsNaN( "hi there!" )
+        lu.assertNotIsNaN( nil )
+        lu.assertNotIsNaN( {} )
+        lu.assertNotIsNaN( {1,2,3} )
+        lu.assertNotIsNaN( {1} )
+        lu.assertNotIsNaN( coroutine.create( function(v) local y=v+1 end ) )
+        assertFailure(lu.assertNotIsNaN, 0 / 0)
+        lu.assertNotIsNaN( 0 / 1 ) -- 0.0
```

Comment:
Same remark as above, let's validate with a few more test cases, just for the sake of completeness.